### ISSUE TITLE
Investigate Uglify

### ISSUE THREAD

**Issue by xjamundx** [TheLarkInn/js-parser-discussions#3]
A lot of modern tools use an esprima/estree-compatible parser.

Can we find out how many projects are using uglify and if it should be included in this discussion.

**Comment by TheLarkInn** [comment]
They consider themself a parser though. And I believe implement AST parsing by hand (no deps) for 1?

**Comment by xjamundx** [comment]
They do, and it's a cool project but it's not ESTree/Esprima compatible from what I understand. It's kind of it's own thing and not really used by any projects outside of minifiers (let's actually find out).

**Comment by TheLarkInn** [comment]
Yeah I think there is lessons learned from it. Pros, cons. Features, etc. Maybe we can update this issue to "Investigate Uglify". Thoughts?

**Comment by xjamundx** [comment]
Updated the OP a bit...

**Comment by TheLarkInn** [comment]
awesome! Thank you.

**Comment by mikesherov** [comment]
I believe both Uglify and JSHint both implemented their own parsers to do streaming analysis rather than a separate parse / analyze / transform step.

**Comment by xjamundx** [comment]
Right and same with jslint :)

**Comment by RReverser** [comment]
UglifyJS actually does have separate parsing (http://lisperator.net/uglifyjs/parser) / compression (http://lisperator.net/uglifyjs/compress) stages, and allows writing own transformers (http://lisperator.net/uglifyjs/transform).
